### ISSUE TITLE
Build issue on Mac?

### ISSUE THREAD

**Issue by jyap808** [neutrinofoundation/neutrino-digital-currency#3]
I get the following error when running qmake "bitcoin-qt.pro" and then "make"

```
/opt/local/bin/uic src/qt/forms/sendcoinsdialog.ui -o build/ui_sendcoinsdialog.h
/opt/local/bin/uic src/qt/forms/coincontroldialog.ui -o build/ui_coincontroldialog.h
/opt/local/bin/uic src/qt/forms/addressbookpage.ui -o build/ui_addressbookpage.h
/opt/local/bin/uic src/qt/forms/signverifymessagedialog.ui -o build/ui_signverifymessagedialog.h
/opt/local/bin/uic src/qt/forms/aboutdialog.ui -o build/ui_aboutdialog.h
/opt/local/bin/uic src/qt/forms/editaddressdialog.ui -o build/ui_editaddressdialog.h
/opt/local/bin/uic src/qt/forms/transactiondescdialog.ui -o build/ui_transactiondescdialog.h
/opt/local/bin/uic src/qt/forms/overviewpage.ui -o build/ui_overviewpage.h
/opt/local/bin/uic src/qt/forms/sendcoinsentry.ui -o build/ui_sendcoinsentry.h
/opt/local/bin/uic src/qt/forms/askpassphrasedialog.ui -o build/ui_askpassphrasedialog.h
/opt/local/bin/uic src/qt/forms/rpcconsole.ui -o build/ui_rpcconsole.h
/opt/local/bin/uic src/qt/forms/optionsdialog.ui -o build/ui_optionsdialog.h
/usr/bin/clang++ -c -pipe -O2 -arch x86_64 -Xarch_x86_64 -mmacosx-version-min=10.8 -Wall -W -isysroot /Developer/SDKs/MacOSX10.8.sdk -DQT_GUI -DBOOST_THREAD_USE_LIB -DBOOST_SPIRIT_THREADSAFE -DUSE_UPNP=1 -DSTATICLIB -DUSE_IPV6=1 -DHAVE_BUILD_INFO -DMAC_OSX -DMSG_NOSIGNAL=0 -DQT_NO_DEBUG -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQT_SHARED -I/opt/local/share/qt4/mkspecs/macx-g++ -Ibuild -I/opt/local/Library/Frameworks/QtCore.framework/Versions/4/Headers -I/opt/local/Library/Frameworks/QtNetwork.framework/Versions/4/Headers -I/opt/local/Library/Frameworks/QtGui.framework/Versions/4/Headers -Isrc -Isrc/json -Isrc/qt -Isrc/tor -Isrc/leveldb/include -Isrc/leveldb/helpers -I/usr/local/include -I/usr/local/opt/berkeley-db4/include -I/opt/local/Library/Frameworks/QtGui.framework/Versions/4/Headers -I/opt/local/Library/Frameworks/QtNetwork.framework/Versions/4/Headers -I/opt/local/Library/Frameworks/QtCore.framework/Versions/4/Headers -I/opt/local/include -Ibuild -F/opt/local/Library/Frameworks -F/opt/local/lib -o build/macdockiconhandler.o src/qt/macdockiconhandler.mm
In file included from src/qt/macdockiconhandler.mm:1:
In file included from src/qt/macdockiconhandler.h:4:
In file included from /opt/local/Library/Frameworks/QtCore.framework/Versions/4/Headers/QObject:1:
In file included from /opt/local/Library/Frameworks/QtCore.framework/Versions/4/Headers/qobject.h:47:
In file included from /opt/local/include/QtCore/qobjectdefs.h:45:
In file included from /opt/local/include/QtCore/qnamespace.h:45:
/opt/local/include/QtCore/qglobal.h:68:10: fatal error: 'algorithm' file not found
#include <algorithm>
         ^
1 error generated.
make: *** [build/macdockiconhandler.o] Error 1
```

**Comment by emilburzo** [comment]
Can you try installing Qt from homebrew?

**Comment by jyap808** [comment]
... Yeah, it's exactly the same error point with homebrew.  Just different paths below.

```
In file included from src/qt/macdockiconhandler.mm:1:
In file included from src/qt/macdockiconhandler.h:4:
In file included from /usr/local/Cellar/qt/4.8.6/lib/QtCore.framework/Versions/4/Headers/QObject:1:
In file included from /usr/local/Cellar/qt/4.8.6/lib/QtCore.framework/Versions/4/Headers/qobject.h:47:
In file included from /usr/local/Cellar/qt/4.8.6/include/QtCore/qobjectdefs.h:45:
In file included from /usr/local/Cellar/qt/4.8.6/include/QtCore/qnamespace.h:45:
/usr/local/Cellar/qt/4.8.6/include/QtCore/qglobal.h:68:10: fatal error: 'algorithm' file not found
#include <algorithm>
         ^
1 error generated.
make: *** [build/macdockiconhandler.o] Error 1
```

**Comment by jyap808** [comment]
I had to comment out these lines to make it work:

```
QMAKE_MACOSX_DEPLOYMENT_TARGET=10.8
QMAKE_MAC_SDK = /Developer/SDKs/MacOSX10.8.sdk
```

Build fine now.
